appstudio-utils/util-scripts/lib/fetch/tekton.sh

```diff
@@ -25,7 +25,7 @@ tr-get-result() {
 
 tr-get-task-name() {
   local tr=$1
-   oc get tr/$tr -o jsonpath="{.spec.taskRef.name}"
+   oc get tr/$tr -o jsonpath="{.metadata.labels['tekton\.dev/pipelineTask']}"
```

Comment:
The value determined here influences the placement of `data.json` file, which in turn influences the `data` structure in OPA. If we used the `.metadata.name` then we can't use logic like this:

https://github.com/hacbs-contract/ec-policies/blob/b1ec19145f4c9fd3b116502b227af7d76463ccb2/policies/test.rego#L25-L26

Reply:
It sounds like referencing the data that way is flawed anyways. Maybe we need to ignore whatever the name of the file is in our policies and simply do some sort of filtering based on a nested attribute.

Reply:
I think something that identifies the origin of the data (now we use a task name) needs to originate from outside of the data itself. We need to assert that the data corresponds to a known task, otherwise someone could inject a task with the data that asserts success and try to subvert the enterprise contract evaluation.

Reply:
Getting the identifier from `.spec` and `.metadata` does not create a different origin for the identifier. IMO, task identifiers such as name are weak. A more resilient approach is to look at different attributes of the task definition, such as `image`, `bundle`, etc. There's a bit of heuristics involved - I don't think we can get away from that.

Reply:
I've created [HACBS-445](https://issues.redhat.com/browse/HACBS-445) to follow up on this.